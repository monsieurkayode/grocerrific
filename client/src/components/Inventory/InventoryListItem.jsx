/* eslint-disable no-underscore-dangle */
import React from 'react';
import { func, string, number } from 'prop-types';

const InventoryListItem = ({
  openModal,
  openDeleteModal,
  _id,
  name,
  quantity,
  price
}) => (
  <div className="inventory__item_wrapper">
    <div className="item-name">{name}</div>
    <div className="item-price">&#8358;{price}</div>
    <div className="item-quantity">{quantity}</div>
    <div className="buttons">
      <button
        onClick={() => openModal({
          id: _id,
          name,
          price,
          quantity,
        })}
        className="edit"
        type="button"
      >
        Edit
      </button>
      <button
        className="delete"
        type="button"
        onClick={() => openDeleteModal({ id: _id, name })}
      >
        Delete
      </button>
    </div>
  </div>
);

InventoryListItem.propTypes = {
  openModal: func.isRequired,
  openDeleteModal: func.isRequired,
  _id: string.isRequired,
  name: string.isRequired,
  price: number.isRequired,
  quantity: number.isRequired
};

export default InventoryListItem;
