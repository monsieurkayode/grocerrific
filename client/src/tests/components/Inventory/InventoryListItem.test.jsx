import React from 'react';
import { shallow } from 'enzyme';
import InventoryListItem from '../../../components/Inventory/InventoryListItem';
import mockGroceries from '../../__mocks__/mockGroceries';

const setup = () => {
  const grocery = mockGroceries[0];
  const props = {
    groceries: mockGroceries,
    openModal: jest.fn(),
    openDeleteModal: jest.fn(),
    loading: false,
    ...grocery
  };

  const wrapper = shallow(<InventoryListItem {...props} />);

  return { wrapper, props };
};

describe('InventoryListItem Component', () => {
  const { wrapper, props } = setup();

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render an Edit and Delete button', () => {
    expect(wrapper.find('.edit').length).toBe(1);
    expect(wrapper.find('.delete').length).toBe(1);
  });

  it('should call openModal when the Edit button is clicked', () => {
    const editButton = wrapper.find('.edit');
    editButton.simulate('click');
    expect(props.openModal).toHaveBeenCalled();
  });

  it('should call openDeleteModal when the Delete button is clicked', () => {
    const deleteButton = wrapper.find('.delete');
    deleteButton.simulate('click');
    expect(props.openDeleteModal).toHaveBeenCalled();
  });
});
