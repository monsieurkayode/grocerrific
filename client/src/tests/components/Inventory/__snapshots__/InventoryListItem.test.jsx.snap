// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InventoryListItem Component renders correctly 1`] = `
<div
  className="inventory__item_wrapper"
>
  <div
    className="item-name"
  >
    mamador
  </div>
  <div
    className="item-price"
  >
    â‚¦
    1250
  </div>
  <div
    className="item-quantity"
  >
    3
  </div>
  <div
    className="buttons"
  >
    <button
      className="edit"
      onClick={[Function]}
      type="button"
    >
      Edit
    </button>
    <button
      className="delete"
      onClick={[Function]}
      type="button"
    >
      Delete
    </button>
  </div>
</div>
`;
